roles/common/tasks/redhat.yml

```diff
@@ -60,7 +60,18 @@
   until: java_install_result is success  or ansible_check_mode
   retries: 10
   delay: 5
-  when: install_java|bool
+  when: install_java|bool and ansible_distribution != "Amazon"
+  tags: package
+
+- name: Install Java
```

Comment:
Maybe this task and above task can be named differently. Something like Install Java on Amazonlinux and Install Java on Rhel